### PR TITLE
vlan buf fix

### PR DESCRIPTION
**Pull request by dragonorloong**
1 set freebsd mbuf vlan information when the vlan_strip is enable 
    2 get ether_type error when the vlan_strip is disabled

### CODE REVIEW

File: lib/ff_dpdk_if.c [hunk 1]
```diff
@@ -97,6 +97,8 @@ static int enable_kni;
 static int kni_accept;
 #endif
 
+#define ETH_P_8021Q 0x8100
```
Comment (whl739) [inline lib/ff_dpdk_if.c:100]: You can use `ETHER_TYPE_VLAN` which defined in `rte_ether.h`.

File: lib/ff_dpdk_if.c [hunk 2]
```diff
@@ -901,15 +902,23 @@ ff_veth_input(const struct ff_dpdk_if_context *ctx, struct rte_mbuf *pkt)
 }
 
 static enum FilterReturn
-protocol_filter(const void *data, uint16_t len)
+protocol_filter(const struct rte_mbuf *mbuf, uint16_t len)
```
Comment (whl739) [inline lib/ff_dpdk_if.c:905]: It seems that this is not necessary to use `mbuf`.

File: lib/ff_dpdk_if.c [hunk 3]
```diff
@@ -901,15 +902,23 @@ ff_veth_input(const struct ff_dpdk_if_context *ctx, struct rte_mbuf *pkt)
...
     if(len < ETHER_HDR_LEN)
         return FILTER_UNKNOWN;
 
     const struct ether_hdr *hdr;
+    const struct vlan_hdr *vlanhdr;
+    void *data = rte_pktmbuf_mtod(mbuf, void*);
     hdr = (const struct ether_hdr *)data;
+    uint16_t ether_type = ntohs(hdr->ether_type);
+
+    if (hdr->ether_type == htons(ETH_P_8021Q) && !(mbuf->ol_flags & PKT_RX_VLAN_STRIPPED)) {
```
Comment (jfb8856606) [inline lib/ff_dpdk_if.c:916]: `ether_type` do twice `ntohs` or `htons`

Comment (jfb8856606) [reply lib/ff_dpdk_if.c:916]: And the second condition is useless.

**Comment by jfb8856606** [comment]
Merge it and will do some modify.

**Pull request merged**